catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CloudMessagingTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.cloudmessaging;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.test.AndroidTestCase;
+
+import org.catrobat.catroid.cloudmessaging.CloudMessaging;
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.junit.After;
+import org.junit.Before;
+
+public class CloudMessagingTest extends AndroidTestCase {
+
+	private Context context;
+	private SharedPreferences.Editor editor;
+
+	@Before
+	public void setUp() {
+		context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		editor = sharedPreferences.edit();
+		editor.clear();
+		editor.putBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, true);
+		editor.commit();
```

Comment:
I think 
`PreferenceManager.getDefaultSharedPreferences(context);` is different from `context.getSharedPreferences('cloud_testing', 0);` as their names are different so different files will be created (values will not be shared) (do correct me if I am wrong)

and with this approach, I was getting an issue that [this preference](https://github.com/nishantt95/Catroid/blob/f0bc32f9378506e671a38c161c1a42986a6df309/catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CloudMessagingTest.java#L45-45) (actual implementation -> context.getSharedPreferences('cloud_testing', 0);) is different from this [sharedPreference](https://github.com/nishantt95/Catroid/blob/f0bc32f9378506e671a38c161c1a42986a6df309/catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java#L79-79) and I was not able to set the sharedPrefernce of CloudMessaging.java from CloudMessagingTest.java.

> SharedPreferences implementation. Just with HashMaps and to get/set values synchronously. Every more complex method will just throw an UnsupportedOperationException.
> We could also provide a method to store and restore the SharedPreferences. This can be used here. Btw, such implementation is officially called a Fake 😉

Implementation part is still not clear as (used [here](https://github.com/Catrobat/Catroid/pull/2377/files#diff-c41b068c0724c4fe613f723e96219ba1R81)) is the normal one and can't figure out that how this one is different from the one we have used normally.